src/components/Modals/AddMovieModal/addmoviemodal.js

```diff
@@ -2,37 +2,93 @@ import React from 'react';
-  <Modal
-    isOpen={show}
-    onRequestClose={onClose}
-    contentLabel="Example Modal"
-    appElement={document.getElementById('app')}
-    className="add-movie-modal"
-  >
-    <div className="close" onClick={onClose}>
-      &times;
-    </div>
-    <form action="/" method="post" className="modal-content">
-      <h2 className="add-movie-header">Add movie</h2>
-      <FormInput title="TITLE" type="text" />
-      <FormInput title="RELEASE DATE" type="date" />
-      <FormInput title="MOVIE URL" type="text" />
-      <div className="form-field-header">GENRE</div>
-      <select className="form-field-input">
-        <option>Comedy</option>
-        <option>Drama</option>
-        <option>Horror</option>
-      </select>
-      <FormInput title="OVERVIEW" type="text" />
-      <FormInput title="RUNTIME" type="text" />
-      <div className="form-buttons">
-        <input className="reset-button" type="reset" value="RESET" />
-        <button className="submit-button" type="submit">SUBMIT</button>
+
+const AddMovieModal = ({ show, onClose }) => {
+  
+  const dispatch = useDispatch();
+  
+  const formik = useFormik({
+    initialValues: {
+      title: '',
+      release_date: '',
+      poster_path: '',
+      genres: 'Comedy',
+      overview: '',
+      runtime: 0,
+    },
+    validationSchema: Yup.object({
+      title: Yup.string()
+        .max(15, 'Must be 15 characters or less')
+        .required('Required'),
+      release_date: Yup.date()
+        .required('Required'),
+      poster_path: Yup.string()
+        .url('Invalid Url')
+        .required('Required'),
+      overview: Yup.string()
+        .max(100, 'Must be 100 characters or less')
+        .required('Required'),
+      runtime: Yup.number()
+        .integer('Must be an integer')
+        .required('Required'),
+    }),
+    onSubmit: (values) => {
+      dispatch(addNewMovie(values));
+      onClose();
+    },
+  });
+  
+  return (
+    <Modal
+      isOpen={show}
+      onRequestClose={onClose}
+      contentLabel="Example Modal"
+      appElement={document.getElementById('app')}
+      className="add-movie-modal"
+    >
+      <div className="close" onClick={onClose}>
+        &times;
       </div>
-    </form>
-  </Modal>
-);
+      <form onSubmit={formik.handleSubmit} className="modal-content">
+        <h2 className="add-movie-header">Add movie</h2>
+        <FormInput title="TITLE" id="title" name="title" type="text" onChange={formik.handleChange} value={formik.values.title} onBlur={formik.handleBlur} />
+        {formik.touched.title && formik.errors.title ? (
+          <div>{formik.errors.title}</div>
+        ) : null}
+        <FormInput title="RELEASE DATE" id="release_date" name="release_date" type="date" onChange={formik.handleChange} value={formik.values.release_date} onBlur={formik.handleBlur} />
+        {formik.touched.release_date && formik.errors.release_date ? (
+          <div>{formik.errors.release_date}</div>
+        ) : null}
```

Comment:
I think that can be extracted to some common component as well since the logic is being repeated

Reply:
done